lib/cron_parser.rb

```diff
@@ -295,4 +304,16 @@ def validate_source
       raise ArgumentError, 'not a valid cronline'
     end
   end
+
+  def strict_match?
+    @strict_match
+  end
+
+  def set_strict_match(strict_match = false)
+    @strict_match = false
+    if strict_match && [time_specs[:dom].last, time_specs[:dow].last].all?{|v| v != '*'}
```

Comment:
Yeah I just notifed when reading through the whole file that this beast is in serious need of a refactoring.

My plan yould be to chunk it up into understandable pieces, like having a `CronParser::Default` and `CronParser::Strict` class. But for that, lots of code would need to be changed, so I would do that after incorporating your changes.